sql/src/main/java/io/crate/operation/scalar/arithmetic/MapFunction.java

```diff
@@ -49,22 +47,26 @@
 
     public final static String NAME = "_map";
 
-    private final static List<Signature> SIGNATURES = ImmutableList.<Signature>builder()
-        .add(new Signature()) // allow empty map
-        .add(new Signature(0, StringType.INSTANCE, AnyType.INSTANCE))
-        .build();
+    private final static FunctionResolver RESOLVER = new FunctionResolvers.BaseFunctionResolver(
+        // emtpy args or (string, any) pairs
+        Signature.of(() -> new Iterator<Signature.ArgMatcher>() {
+            private int pos = 0;
 
-    private final static FunctionResolver RESOLVER = new FunctionResolver() {
+            @Override
+            public boolean hasNext() {
+                return true;
+            }
+
+            @Override
+            public Signature.ArgMatcher next() {
+                return pos++ % 2 == 0 ? Signature.ArgMatcher.STRING : Signature.ArgMatcher.ANY;
```

Comment:
maybe move the pos incremention in an extra line, to make the order of execution more explicit